node/controllers/channelInvitation/index.js

```diff
@@ -45,105 +40,146 @@ export const getChannelInvitations = async (req, res) => {
       },
     ],
   })
-    .then(results => {
-      return results[0].interests
+    .then(result => {
+      return result.interests
     })
     .catch(err => {
       return flatted.stringify(err)
     })
 
-  const channelInvitations = await Promise.all([
-    channelsPromise,
-    userInterestsPromise,
-  ])
-    .then(results => {
-      const { data } = results[0]
-      const interests =
-        results[1] && results[1].map(interest => interest.dataValues.name)
-      const found =
-        data && data.filter(channel => interests.includes(channel.display_name))
-      return [found, interests]
-    })
-    .catch(err => {
-      flatted.stringify(err)
-      return []
-    })
+  // Make a promise that returns all channels
+  const channelsPromise = axios.get(
+    `${mattermostUrl}/teams/${process.env.TEAM_ID}/channels`
+  )
 
   try {
-    const channelsPromises =
-      channelInvitations[0] &&
-      (await Promise.all(
-        channelInvitations[0].map(channel => {
-          return axios.get(`${mattermostUrl}/channels/${channel.id}/stats`)
-        })
-      ))
-
-    const channelsWithRoom = channelsPromises
-      .filter(result => result.data.member_count < 8)
-      .map(result => result.data.channel_id)
-
-    let exhaustedInterests = []
-
-    const found = channelInvitations[0].filter(channel => {
-      if (
-        channelsWithRoom.includes(channel.id) &&
-        !exhaustedInterests.includes(channel.display_name)
-      ) {
-        exhaustedInterests = [...exhaustedInterests, channel.display_name]
-        return channelsWithRoom.includes(channel.id)
-      }
-      return false
+    // Wait till both promises are done, and map values into stuff we want
+    // interestsPromise becomes a list that just contains calling users interests
+    // channelsPromise becomes a list containing all channel stats
+    // we also add channel member count into returned channel data (it is not normally in channels, but in channels stats)
+    // Here we also use await and .then because we want data to contain the values returned from .then
```

Comment:
This works fine and I totally get your point, but my personal preference is still to stick with either .then chaining or async/awaits and not mix those. I would maybe write this just using awaits like this:

```
const results = await Promise.all([userInterestsPromise, channelsPromise])
const interests = results[0] && results[0].map(interest => interest.name)

const channelsData =
    results[1].data &&
    results[1].data
      .filter(
        channelData =>
          channelData.name !== 'town-square' && channelData.name !== 'off-topic'
      )
      .map(async channelData => {
        const newChannelData = { ...channelData }
        const channelStats = await axios.get(
          `${mattermostUrl}/channels/${channelData.id}/stats`
        )
        // Add the membercount variable to channelData
        // The given parameter was fetched for this purpose and is disregarded after, so we do not care if we modify it here.
        newChannelData.memberCount = channelStats.data.member_count
        // Also parse channel purpose to js object
        newChannelData.purpose =
          channelData && channelData.purpose && JSON.parse(channelData.purpose)
        return newChannelData
      })
      
const data = [interests, channelsData]
```

IMO it's easier to read. What do you think?

Reply:
And similarly elsewhere